### PR TITLE
feat: vendor management

### PR DESCRIPTION
**Pull request by tshegofatsoseane**
**Description** 
Added CRUD routes for vendor/supplier management
 
**Changes**
- Added controller for vendor/supplier management
- Added model for vendor/supplier management
- Added routes for vendor/supplier management
- Modified app.js to include vendor/supplier management routes
- Added tests

Create new vendor `POST http://localhost:8080/vendors/`
Screenshot 2024-04-30 at 01 44 02

Retrieve all vendors `GET http://localhost:8080/vendors/`
Screenshot 2024-04-30 at 01 44 25

Get vendor by ID `GET http://localhost:8080/vendors/6630260256458f325c531760`
Screenshot 2024-04-30 at 01 44 43

**Tests**
Screenshot 2024-04-30 at 01 38 34

This addresses issue #17

### CODE REVIEW

File: backend/controllers/vendorController.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+const Vendor = require('../models/vendor');
+
+// list all vendors
+exports.getAllVendors = async (req, res) => {
+  try {
+    const vendors = await Vendor.find();
```
Comment (khumodiatla) [inline backend/controllers/vendorController.js:6]: Add if vendors.length === 0  check

File: backend/controllers/vendorController.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+exports.createVendor = async (req, res) => {
+  try {
+    const { name, email, phone } = req.body;
+    
+    if (!name || !email || !phone) {
+      return res.status(400).json({ message: 'Name, email, and phone are required' });
+    }
+
+    const vendor = new Vendor({ name, email, phone });
+    await vendor.save();
```
Comment (khumodiatla) [inline backend/controllers/vendorController.js:22]: Add if vendor already exists check before creating one

File: backend/models/vendor.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+const mongoose = require('mongoose');
+
+const vendorSchema = new mongoose.Schema({
+  name: {
+    type: String,
+    required: true
+  },
+  email: {
+    type: String,
+    required: true
```
Comment (khumodiatla) [inline backend/models/vendor.js:10]:
```suggestion
  email: {
    type: String,
    required: true,
    unique: true
```

File: backend/controllers/vendorController.js (hunk 2 above)
Comment (tshegofatsoseane) [reply backend/controllers/vendorController.js:36]: Added them

File: backend/controllers/vendorController.js (hunk 1 above)
Comment (tshegofatsoseane) [reply backend/controllers/vendorController.js:6]: Added

File: backend/controllers/paymentController.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
+const Payment = require('../models/Payment');
+
+exports.getAllPayments = async (req, res) => {
+  try {
+    const payments = await Payment.find();
```
[the remark this answers is not in the document]
Comment (tshegofatsoseane) [reply backend/controllers/paymentController.js:5]: done

**Review by khumodiatla — APPROVED** [review]

**Pull request merged**